### ISSUE TITLE
Issues when upgrading to 0.6.0

### ISSUE THREAD

**Issue by Beatso** [TheMrZZ/sandstone#38]
When I upgraded sandstone to 0.6.0 in my project, I got this error message.
<details>
<summary>console error message</summary>
> npx ts-node index.ts

Error: Cannot define namespace under a base path.
    at BasePathClass.getName (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\src\_internals\datapack\BasePath.ts:49:13)  
    at Object.BasePathClass.MCFunction (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\src\_internals\datapack\BasePath.ts:137:64)
    at Object.<anonymous> (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\index.ts:27:1)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Module.m._compile (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:1043:23)      
    at Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Object.require.extensions.<computed> [as .ts] (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:1046:12)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! manhunt@1.0.0 build: `npx ts-node index.ts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the manhunt@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Beatso\AppData\Roaming\npm-cache\_logs\2020-11-29T13_33_24_088Z-debug.log
</details>

After replacing all `mcfunction`s with `MCFunction`, i got this error:
<details>
<summary>console error message</summary>
> npx ts-node index.ts

C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:500
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
index.ts:2:22 - error TS2305: Module '"./node_modules/sandstone/core"' has no exported member 'saveDatapack'.

2 import { MCFunction, saveDatapack } from "sandstone/core"
                       ~~~~~~~~~~~~

    at createTSError (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:500:12)
    at reportTSError (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:504:19)
    at getOutput (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:739:36)
    at Object.compile (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:955:32)
    at Module.m._compile (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:1043:43)      
    at Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Object.require.extensions.<computed> [as .ts] (C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks\SurvivalistsVsHitmen\node_modules\ts-node\src\index.ts:1046:12)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! manhunt@1.0.0 build: `npx ts-node index.ts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the manhunt@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Beatso\AppData\Roaming\npm-cache\_logs\2020-11-29T13_34_49_146Z-debug.log
</details>

what am I doing wrong, and how can I fix this issue? is this an issue with the release of sandstone, or what I'm doing?
[link to my code](https://github.com/Beatso/SurvivalistsVsHitmen/blob/update-sandstone-0.6.0/index.ts)

**Comment by TheMrZZ** [comment]
Hello,

Indeed, Sandstone v0.6.0 changed a lot of things. This isn't documented yet (sorry for that), but to use Sandstone v0.6.0 you will probably want to create a whole new project (don't worry, the code stays identical, but the structure of the project changed).

Basically, Sandstone v0.6.0 introduces a CLI, which helps to create/build/watch projects. This is a beta CLI for the moment, and is not battle-tested. It adds support for a configuration file, automatic saving, automatic import of your files etc... 

Therefore, you have 2 options:
- Fix a bunch of stuff and stay on the "manual" version
- Try the new, beta CLI and upgrade to the "automatic" version

If you want to stay on the "manual" one, you will need to import `savePack` instead of `saveDatapack`. This change is to be future-proof, since Sandstone will probably support resource packs in the long run. You also need to upgrade to Sandstone v0.6.2, which fixes the BasePath-related bug (thanks for letting me know btw, you're the first to report this bug!)

If you want to upgrade to the CLI, you'll have to create a new project. You will get automatic saving, neat configurations, and a nice CLI that is way better than running `ts-node index.ts`! Here are the steps:
1. Go to a root folder, like `projects`. A sub folder will be created for your Sandstone project.
2. Run `npx sandstone-cli create my-project-name`. Change my-project-name to your project's name: it **will not** be the name of the data pack, just the name of the folder, so don't worry
3. Answer all the questions. All your answers can be changed later in the configuration file, so if you make a mistake don't worry!
4. Open your new projects, that's it!

To build, you can run `npm run build`. To build & rebuild on each change, you can run `npm run watch`.
Once again, it's a very beta CLI, so expect a few bug. Report them & I'll fix them as soon as possible!

I'll keep this issue opened for a bit if you've got any problem, then I'll close it. I'm interested in getting your feedback on Sandstone btw: add me on Discord (TheMrZZ#9307), or drop by the [Minecraft Commands Discord](https://www.reddit.com/r/MinecraftCommands/comments/48u784/discord_community_server_for_rminecraftcommands/) and tag me if you want!

**Comment by Beatso** [comment]
@TheMrZZ Thanks for your quick reply! Here's what I ran into:

Using the manual method, I was already using `savePack` (since saveDatapack didn't work for me in v0.5.4) when I got the errors, but updating to v6.0.2 did the trick, thanks :).

When using the CLI, I got the following error:
```
C:\Users\Beatso\OneDrive\Documents\Minecraft\datapacks>npx sandstone-cli create survivalists-vs-hitmen
npx: installed 111 in 23.011s
? Name of your data pack (can be changed later) > Survivalists Vs Hitmen
? Where do you want your datapack to be saved (can be changed later)? .minecraft/datapacks folder
? Default namespace (can be changed later) > hitman_internal
? What package manager do you want to use? > npm
Installing sandstone, sandstone-cli and typescript using npm.
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN survivalists-vs-hitmen@1.0.0 No description
npm WARN survivalists-vs-hitmen@1.0.0 No repository field.

npm WARN survivalists-vs-hitmen@1.0.0 No description
npm WARN survivalists-vs-hitmen@1.0.0 No repository field.

    Error: ENOENT: no such file or directory, stat 'C:\Users\Beatso\AppData\Roaming\npm-cache\_npx\6860\node_modules\sandstone-cli\lib\template'      
    Code: ENOENT
```

**Comment by TheMrZZ** [comment]
No problem! I'm happy to help :)  
The error you got is fixed. I tried with the new package & everything is working fine, can you try it?

Just recreate a project with `npx sandstone-cli create my-project-name`. If you get this `ENOENT` error once again, it means `npx` didn't notice there was a new version of `sandstone-cli`.
In that case, you might want to run `npx sandstone-cli@0.1.4 create my-project-name` to force him to update. But please first try without specifying the version (it will help me in future bugfix, to know whether `npx` updates automatically or if I should always tell people to specify the version).

Once again, keep me informed! I hope you'll be able to try this out without any bug.

**Comment by Beatso** [comment]
Just tried exactly the same thing since you released v0.1.4. I didn't need to specify the version, it updated automatically.

Just also wanted to say thanks so much for creating this library, and your help here :)

**Comment by TheMrZZ** [comment]
So the bug is fixed? Perfect!

I'm happy to hear it, thanks 🙂 Once the documentation is over, I'll officially release it & I hope it will get some traction!

Once again, feel free to contact me on Discord if you need to. I'll close this issue for now, have a nice week!
